### PR TITLE
feat(refactor): Move sub get_number_of_male_reads

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR adds | fixes:

- See above

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [ ] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: lib/MIP/Parse/Gender.pm [hunk 1]
```diff
@@ -18,87 +17,25 @@ use Readonly;
...
-            commands_ref => $cmds_ref,
-            process_type => q{ipc_cmd_run},
-        }
-    );
-
-    my $y_read_count = $process_return{stdouts_ref}->[0];
-
-    ## Clean-up
-    close $filehandle;
-    remove_tree($bash_temp_file);
```
Comment (jemten) [inline lib/MIP/Parse/Gender.pm:97]: Just curious. Why do we delete the file after we're done? Also, we could add `tee` to the command in order to save the output of the command to a file as well. But again the better option here is probably to rewrite this module a bit in order to log some more values and what command has been run. Another PR :)

**Review by jemten — APPROVED** [review]
👍

File: lib/MIP/Parse/Gender.pm (hunk 1 above)
Comment (henrikstranneheim) [reply lib/MIP/Parse/Gender.pm:97]: Yeah, I also wondered that. It seems wrong as one removes the record of what was done making it hard to reproduce and inspect

**Comment by henrikstranneheim** [comment]
@jemten I made some changes - could you give it another look?

**Review by jemten — APPROVED** [review]
Nice! Now we have a track record of the command that was run

File: lib/MIP/Parse/Gender.pm (hunk 1 above)
Comment (henrikstranneheim) [reply lib/MIP/Parse/Gender.pm:97]: Fixed, except tee suggestion

**Pull request merged**